mesonbuild/backend/ninjabackend.py

```diff
@@ -1376,6 +1379,33 @@ def generate_phony(self) -> None:
         elem = NinjaBuildElement(self.all_outputs, 'PHONY', 'phony', '')
         self.add_build(elem)
 
+    def generate_shader_target(self, target: build.Shader):
+        fname = target.get_filename()
+        outname = os.path.join(self.get_target_dir(target), fname)
+        compiler = target.compilers['glsl']
+        sources = target.get_sources()
+        src = []
+        for s in sources:
+            src.append(s.rel_to_builddir(self.build_to_src))
+
+        # Add possible shader generated files to src list
+        generated_sources = self.get_target_generated_sources(target)
+        gen_src_list = []
+        for rel_src in generated_sources.keys():
+            raw_src = File.from_built_relative(rel_src)
+            gen_src_list.append(raw_src)
+
+        args = self.generate_basic_compiler_args(target, compiler)
+        args += compiler.get_output_args(outname)
```

Comment:
For a compiler, we typically know a lot about how it works and can write a targeted, optimized build rule. get_output_args is usually placed inside the rule definition itself, since it's universal to all invocations of the rule.